src/inmanta/ast/type.py

```diff
@@ -237,7 +237,7 @@ class Number(Primitive):
 
     def __init__(self) -> None:
```

Comment:
the docstring on the class is not entirely correct

Reply:
In my other [PR](https://github.com/inmanta/inmanta-core/pull/6686/files#diff-0a38a04c81c27002daab3fea8b2c4eac57e25c020dcca78a124df914b5de6ab3R232) I changed it like this so I've just made the same change here. I think it is indeed just a float now. That's the plan right @sanderr ?

Reply:
This is a good point. For iso6 we probably want `number(1)` to be stable and produce an int. So for iso7 I like this as is, for iso6 perhaps we have to override the cast method so it does an `int` cast if the value is already an int and a float cast otherwise.

I hadn't realized we'd want different behavior between iso6 and iso7 here. In this case it's probably easiest to just make this an iso6 PR and to put the iso7 part + bugfix entry in your in progress one.